test/framework/src/main/java/org/elasticsearch/test/InternalAggregationTestCase.java

```diff
@@ -166,7 +166,7 @@
 import static org.hamcrest.Matchers.hasSize;
 import static org.hamcrest.Matchers.lessThanOrEqualTo;
 
-public abstract class InternalAggregationTestCase<T extends InternalAggregation> extends AbstractWireSerializingTestCase<T> {
+public abstract class InternalAggregationTestCase<T extends InternalAggregation> extends AbstractNamedWriteableTestCase<T> {
```

Comment:
Could this be `extends AbstactNamedWriteableTestCase<InternalAggregation>`?

Reply:
Probably, i'll check!

Reply:
I think it needs to remain `<T>`... `AbstractWireTestCase#mutateInstance()` takes the `T`, so if we change it to `<InternalAggregation>` it forces signatures like

```
protected InternalBoxplot mutateInstance(InternalBoxplot instance) {
```

to be

```
protected InternalAggregation mutateInstance(InternalAggregation instance) {
```
and requires a bunch of extra casting.

Reply:
Got it!